### PR TITLE
Chore/1128 hotswap carbonmark api restore git history

### LINKED ISSUE #1128: [api] Hot-swap API deployments
Plan of action here

https://github.com/KlimaDAO/klimadao/pull/1072#pullrequestreview-1417345608

EDIT:
adding to the above:
-- first only swap the staging domain, so that staging is running on the new API deployment, for testing purposes. Atmos will take care of this.

### PR DESCRIPTION
**Pull request by 0xemc**
## Description

This PR brings in the git history from the [KlimaDAO/carbonmark-api](https://github.com/KlimaDAO/carbonmark-api)repository as well as adds a postman collection for documentation and testing and fixes a couple of small bugs.

The postman tests are also now a blocking step as part of the vercel build process:

image

For now they are only checking for 200 status codes but will be more comprehensive later.

## Related Ticket

Resolves #1128

## Checklist

- [x] I have run `npm run build-all` without errors
- [x] I have formatted JS and TS files with `npm run format-all`

## Remaining
- [ ] Re-assign vercel settings to the new domain 
- [ ] verify, add & update user endpoints are not tested in the postman collection (due to the need for message signing with ethers)

### CODE REVIEW

**Comment by 0xemc** [comment]
@Atmosfearful @ladytrekker @Najada 

Unsure why but deployment via the CLI works fine with the current vercel project settings:

https://klimadao-carbonmark-api.vercel.app/api/categories

https://klimadao-carbonmark-8ty3f9i3s-klimadao.vercel.app/api/categories

However the PR github checks are failing and ignoring the vercel project settings defined for the new api project. 

Any ideas?

**Comment by 0xemc** [comment]
> @Atmosfearful @ladytrekker @Najada
> 
> Unsure why but deployment via the CLI works fine with the current vercel project settings:
> 
> https://klimadao-carbonmark-api.vercel.app/api/categories
> 
> https://klimadao-carbonmark-8ty3f9i3s-klimadao.vercel.app/api/categories
> 
> However the PR github checks are failing and ignoring the vercel project settings defined for the new api project.
> 
> Any ideas?

Ok we finally got there.. I had to remove the project from the npm workspaces (similar to carbon-projects) so that a package-lock.json existed. For some reason the install command setting is ignored in the default magic Vercel github PR deployment :(

Now adding some postman tests

**Comment by 0xemc** [comment]
@Atmosfearful please make sure to squash this PR before merging through GH (i'd hate to pollute the history on the staging branch)

File: carbonmark-api/postman_tests.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# Get the process ID of the server
+server_pid=$!
+
+# Wait for the server to be available
+while ! curl -s http://localhost:3000 > /dev/null; do
+  sleep 1
+done
+
+# Run the Newman tests
+npx newman run postman_collection.json
```
Comment (Atmosfearful) [inline carbonmark-api/postman_tests.sh:15]: Amazing, I've never written tests with postman, I think i'll call on you to give a little workshop for us 👑

File: carbonmark-api/routes/projects.js [hunk 2]
```diff
@@ -45,50 +46,70 @@ module.exports = async function (fastify, opts) {
     handler: async function (request, reply) {
       var { country, category, search, vintage } = request.query;
 
+      const [categories, countries, vintages] = await Promise.all([
```
Comment (Atmosfearful) [inline carbonmark-api/routes/projects.js:49]: ⚡

**Review by Atmosfearful — APPROVED** [review]

**Pull request merged**